### PR TITLE
#30279 - Event detection UI and QR updates

### PR DESCRIPTION
**Pull request by srosam**
For sc ticket# 30279

### CODE REVIEW

File: nodejs/advanced/event-detection-ui/library.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		"SDK Usage": ["Read SDK"]
+	},
+	"shortDescription": "An Angular UI for the event detection demo",
+	"DefaultFile": "src/app/services/environment-variables.service.ts",
+	"EntryPoint": "src/build/dockerfile",
+	"Variables": [
+		{
+			"Name" : "topic",
+			"Type" : "EnvironmentVariable",
+			"InputType" : "FreeText",
```
Comment (tomas-quix) [inline nodejs/advanced/event-detection-ui/library.json:16]: Any reason this is not InputTopic?

File: nodejs/advanced/event-detection-ui/library.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+			"Description" : "The topic to read telementry data from",
+			"DefaultValue" : "phone-data",
+			"Required": true
+		},
+		{
+			"Name" : "eventTopic",
+			"Type" : "EnvironmentVariable",
+			"InputType" : "FreeText",
```
Comment (tomas-quix) [inline nodejs/advanced/event-detection-ui/library.json:24]: Any reason this is not InputTopic?

File: nodejs/advanced/event-detection-ui/library.json (hunk 1 above)
Comment (SteveRosam) [reply nodejs/advanced/event-detection-ui/library.json:16]: it makes the pipeline view look bad.
we did this on the sentiment analysis ui too.

File: nodejs/advanced/event-detection-ui/library.json (hunk 2 above)
Comment (SteveRosam) [reply nodejs/advanced/event-detection-ui/library.json:24]: it makes the pipeline view look bad.
we did this on the sentiment analysis ui too.

File: nodejs/advanced/event-detection-ui/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+## Docs
+
+Check out the [API docs](https://docs.quix.io/apis/streaming-reader-api/intro/) for detailed API usage guidance and how to connect to the WebSockets API using SignalR.
+
+## How to run in the Quix Portal
+
+Create an account on [Quix](https://portal.platform.quix.ai/self-sign-up?xlink=github) to edit or deploy this application without a local environment setup.
+
+### Local development
```
Comment (tbedford) [inline nodejs/advanced/event-detection-ui/README.md:13]: Wasn't sure this should be an h3? I think an h2 is fine, as it is not a sub-topic from what I can tell.

File: nodejs/advanced/event-detection-ui/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+To run this project locally please update the following values in `environment-variables.service.ts`
```
Comment (tbedford) [inline nodejs/advanced/event-detection-ui/README.md:15]: Colon before break for code block.

File: nodejs/advanced/event-detection-ui/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+``` 
+  private workingLocally = false; // set to true if working locally
+  public token: string = ''; // Create a token in the Tokens menu and paste it here
+  public topic: string = ''; // Create a topic in the Topic tab - for this demo use `demo-data`
+  public eventTopic: string = ''; // Create a topic in the Topic tab - for this demo use `events`
+  public workspaceId: string = ''; // Look in the URL for the Quix Portal your workspace ID is after `workspace=`
+```
+  
+The Token can be obtained from the Quix portal under the Token menu item (top right of the page under your user avatar)
```
Comment (tbedford) [inline nodejs/advanced/event-detection-ui/README.md:25]: Should this line and the two below be part of a numbered list?

**Review by tbedford — CHANGES_REQUESTED** [review]
A few very minor comments. 

I also notice a directory is specified with spaces: `nodejs/advanced/QR Settings Share/ui/tsconfig.spec.json`. Just checking that won't cause issues?

**Comment by SteveRosam** [comment]
> nodejs/advanced/QR Settings Share/ui/tsconf

files paths with spaces are fine here. the solutions builds. I have had it available and working in the dev library

File: nodejs/advanced/event-detection-ui/README.md (hunk 4 above)
Comment (SteveRosam) [reply nodejs/advanced/event-detection-ui/README.md]: Took me a moment to work out whether you meant there was a colon that shouldn't be there or there wasn't a colon that should be there.. think I got there in the end.

File: nodejs/advanced/event-detection-ui/README.md (hunk 5 above)
Comment (SteveRosam) [reply nodejs/advanced/event-detection-ui/README.md:25]: Changed it slightly.

**Review by tbedford — APPROVED** [review]
LGTM, although I've not tested the code.

**Pull request merged**